### PR TITLE
Platform revision

### PR DESCRIPTION
**Pull request by markoskandylis**
*Issue #, if available:*

*Description of changes:*
- Updating modules versions
- Using pod identity instead of IRSA where we can
- Using external secret to register spokes to hub
- updated hub instances to m5
- Using parameter store to trade info like argo tole beteen hub and spoke

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gitops/addons/bootstrap/aws/addons-aws-oss-external-secrets-appset.yaml [hunk 1]
```diff
@@ -55,8 +55,6 @@ spec:
             values: |
               serviceAccount:
                 name: {{metadata.annotations.external_secrets_service_account}}
-                annotations:
```
Comment (allamand) [inline gitops/addons/bootstrap/aws/addons-aws-oss-external-secrets-appset.yaml:58]: Why you remove the IRSA annotations ?

File: terraform/hub/README.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+Before you begin, make sure you have the following command line tools installed:
+- git
+- terraform
+- kubectl
+- argocd
+
+## Fork the Git Repositories
+
+### Fork the Addon GitOps Repo
+1. Fork the git repository for addons [here](https://github.com/gitops-bridge-dev/gitops-bridge-argocd-control-plane-template).
```
Comment (allamand) [inline terraform/hub/README.md:18]: Why adding this instructions that are for the GitOps bridge repo ?

File: terraform/hub/main.tf [hunk 3]
```diff
@@ -187,23 +195,23 @@ resource "kubernetes_secret" "git_secrets" {
...
 }
-
 ################################################################################
 # GitOps Bridge: Bootstrap
 ################################################################################
 module "gitops_bridge_bootstrap" {
-  source  = "gitops-bridge-dev/gitops-bridge/helm"
-  version = "0.0.1"
+  source = "gitops-bridge-dev/gitops-bridge/helm"
+
```
Comment (allamand) [inline terraform/hub/main.tf:203]: Why removing version?

File: terraform/hub/main.tf [hunk 4]
```diff
@@ -216,53 +224,12 @@ module "gitops_bridge_bootstrap" {
   depends_on = [kubernetes_secret.git_secrets]
 }
 
-################################################################################
-# ArgoCD EKS Access
```
Comment (allamand) [inline terraform/hub/main.tf:220]: We do not need iam role for ArgoCD acccess ?

File: terraform/hub/main.tf [hunk 5]
```diff
@@ -298,37 +268,41 @@ module "eks_blueprints_addons" {
...
+  
   vpc_id     = module.vpc.vpc_id
   subnet_ids = module.vpc.private_subnets
 
+  enable_cluster_creator_admin_permissions = true
+
   eks_managed_node_groups = {
     initial = {
-      instance_types = ["t3.medium"]
+      instance_types = ["m5.large"]
```
Comment (allamand) [inline terraform/hub/main.tf:291]: 👍

File: terraform/hub/manifets/clusterStore.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+apiVersion: external-secrets.io/v1beta1
+kind: ClusterSecretStore
+metadata:
+  name: aws-secrets-manager
+  namespace: external-secrets
+spec:
+  provider:
+    aws:
+      service: SecretsManager
+      region: eu-west-1
```
Comment (allamand) [inline terraform/hub/manifets/clusterStore.yaml:10]: This should be a parameter.

‘Can we avoid using plain manifest and using a helm chart ?

File: terraform/hub/manifets/secretStorev2-prod.yaml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        literal: "cluster_name: '{{ .cluster_name }}'"
+      - target: Labels
+        literal: "environment: '{{ .environment  }}'"
+      data:
+        config: "{{ .config  }}"
+        name:   "{{ .cluster_name  }}"
+        server: "{{ .server  }}"
+  dataFrom:
+    - extract:
+        key: hub-cluster/spoke-prod
```
Comment (allamand) [inline terraform/hub/manifets/secretStorev2-prod.yaml:33]: Where comes from this key ?

File: terraform/hub/pod-identity.tf [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+  cluster_name    = module.eks.cluster_name
+  namespace       = "argocd"
+  service_account = "argocd-server"
+  role_arn        = aws_iam_role.argocd_hub.arn
+}
+
+
+
+################################################################################
+# ESO EKS Access
```
Comment (allamand) [inline terraform/hub/pod-identity.tf:56]: What is ESO ?

File: terraform/spokes/README.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+### Fork the Addon GitOps Repo
+1. Fork the git repository for addons [here](https://github.com/gitops-bridge-dev/gitops-bridge-argocd-control-plane-template).
+2. Update the following environment variables to point to your fork by changing the default values:
+```shell
+export TF_VAR_gitops_addons_org=https://github.com/gitops-bridge-dev
+export TF_VAR_gitops_addons_repo=gitops-bridge-argocd-control-plane-template
+```
+
+### Fork the Workloads GitOps Repo
+1. Fork the git repository for this pattern [here](https://github.com/gitops-bridge-dev/gitops-bridge)
```
Comment (allamand) [inline terraform/spokes/README.md:26]: Same here the repo does not match

File: terraform/spokes/README.md [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+Access ArgoCD's UI, run the command from the output:
+```shell
+terraform output -raw access_argocd
+```
+
+
+## Deploy the Workloads
+Deploy a sample application located in [k8s/game-2048.yaml](k8s/game-2048.yaml) using ArgoCD:
+```shell
+kubectl apply -f bootstrap/workloads.yaml
```
Comment (allamand) [inline terraform/spokes/README.md:116]: This is not relevant for the workshop

File: terraform/spokes/main.tf [hunk 11]
```diff
@@ -146,102 +153,36 @@ locals {
     }
   )
 
-
-  azs = slice(data.aws_availability_zones.available.names, 0, 3)
-
   tags = {
     Blueprint  = local.name
-    GithubRepo = "github.com/csantanapr/terraform-gitops-bridge"
+    GithubRepo = "github.com/gitops-bridge-dev/gitops-bridge"
```
Comment (allamand) [inline terraform/spokes/main.tf:158]: Should be this repo

File: terraform/spokes/main.tf [hunk 12]
```diff
@@ -380,10 +388,37 @@ module "eks" {
           # Reference docs https://docs.aws.amazon.com/eks/latest/userguide/cni-increase-ip-addresses.html
           ENABLE_PREFIX_DELEGATION = "true"
           WARM_PREFIX_TARGET       = "1"
-        }
+        },
+        enableNetworkPolicy = "true"
       })
     }
+    aws-ebs-csi-driver = {
```
Comment (allamand) [inline terraform/spokes/main.tf:395]: We should add Cloudwatch container insights also

**Review by allamand — CHANGES_REQUESTED** [review]

File: gitops/addons/bootstrap/aws/addons-aws-oss-external-secrets-appset.yaml (hunk 1 above)
Comment (markoskandylis) [reply gitops/addons/bootstrap/aws/addons-aws-oss-external-secrets-appset.yaml:58]: Using Pod Identity for the external secret so the IRSA annotation is not needed there

File: terraform/hub/README.md (hunk 2 above)
Comment (markoskandylis) [reply terraform/hub/README.md:18]: Good point i kept the old readme i will remove that

File: terraform/hub/pod-identity.tf (hunk 8 above)
Comment (markoskandylis) [reply terraform/hub/pod-identity.tf:56]: External Secrets Operator (https://external-secrets.io/latest/) ESO :)

File: terraform/spokes/main.tf (hunk 12 above)
Comment (markoskandylis) [reply terraform/spokes/main.tf:395]: Yes this part will be updated on second revision after the core changes are safe

File: terraform/hub/manifets/clusterStore.yaml (hunk 6 above)
Comment (markoskandylis) [reply terraform/hub/manifets/clusterStore.yaml:10]: Yes those will be updated on the second revision we will make incremental changes first platform then gitops appsets

File: terraform/hub/manifets/secretStorev2-prod.yaml (hunk 7 above)
Comment (markoskandylis) [reply terraform/hub/manifets/secretStorev2-prod.yaml:33]: Those are created from the spoke cluster - creating a secret that the hub is pulling

File: gitops/addons/bootstrap/aws/addons-aws-oss-external-secrets-appset.yaml (hunk 1 above)
Comment (allamand) [reply gitops/addons/bootstrap/aws/addons-aws-oss-external-secrets-appset.yaml:58]: Clear thanks

**Pull request merged**